Contents/Code/__init__.py

```diff
@@ -225,13 +228,18 @@ def getStreamItems():
 						]
 					)
 
-				if int(height_s) < best_height or float(fps_s) < best_fps:
+				if Prefs['quality'] == 'Auto':
+					if int(height_s) < best_height or float(fps_s) < best_fps:
+						objects.append(media_object)
+					else:
+						best_height = int(height_s)
+						best_fps = float(fps_s)
+						objects.insert(0, media_object)
+				elif Prefs['quality'] == '720p60' and float(fps_s) == 59.94:
 					objects.append(media_object)
-				else:
-					best_height = int(height_s)
-					best_fps = float(fps_s)
-					objects.insert(0, media_object)
-
+				elif Prefs['quality'] == media_object.video_resolution+'p':
```

Comment:
What if a specific, configured quality isn't available for the stream?
Then the video list will be empty?
There should be a check for this and if preferred quality isn't available, list all others instead.